src/gslc/jsonAssembly.fs

```diff
@@ -0,0 +1,89 @@
+﻿module jsonAssembly
+
+open System.IO
+open System.Text
+open System
+open commonTypes
+open constants
+open Amyris.Bio.utils
+open parseTypes
+open shared
+open Newtonsoft.Json
+open System.Collections.Generic
+
+/// A few key fields of a DNA slice for JSON representation.
+type DNASliceJson =
+   {id: string; 
+    extId: string; 
+    dna: string;  
+    sourceChr: string; 
+    sourceFr: string; 
+    sourceTo: string; 
+    sourceFwd: bool;
+    destFr: string; 
+    destTo: string;
+    destFwd: bool; 
+    amplified: bool; 
+    sliceName: string;
+    sliceType: string;
+    breed: string;
+    description: string ; 
+}
+
+/// A few key fields of Assembly output for JSON representation.
+type AssemblyOutJson = 
+    { id: string;
+    name: string;
+    dnaSlices: DNASliceJson list;
+}
+
+///  Write out a JSON file representing the output assembly list to a given path. 
+let dumpJsonAssemblies (outFile:string) (assemblies : AssemblyOut list) =
+
+    use outF = new StreamWriter(outFile)
+    let assemblyHash =
+        assemblies 
+            |> List.map(fun a ->
+            { id = a.id.Value.ToString(); name = a.name.ToString();
+            dnaSlices = (a.dnaParts |> List.map(fun d ->
+            { id = (match d.id with | None -> "0" | _ -> d.id.Value.ToString());
```

Comment:
Please indent the anonymous function body.  Also, no need for a wildcard pattern here, just explicitly bind the Some case:
```fsharp
match d.id with | None -> "0" | Some(id) -> id.Value.ToString()
```
This is less error-prone as it explicitly binds the result of the match rather than repeating the match clause.